### PR TITLE
Handle new `StoreKitError.notEntitled`

### PR DESCRIPTION
**Pull request by NachoSoto**
Fixes [CF-321]

[CF-321]: https://revenuecats.atlassian.net/browse/CF-321?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by aboedo — APPROVED** [review]

File: Purchases/Error Handling/StoreKitError+Extensions.swift [hunk 1]
```diff
@@ -31,6 +31,11 @@ extension StoreKitError {
         case .notAvailableInStorefront:
             return ErrorUtils.productNotAvailableForPurchaseError(error: self)
 
+#if swift(>=5.6)
+        case .notEntitled:
+            return ErrorUtils.storeProblemError(error: self)
+#endif
+
```
[the remark this answers is not in the document]
Comment (NachoSoto) [reply Purchases/Error Handling/StoreKitError+Extensions.swift:38]: We don't. Surprisingly even the `.userCancelled` path isn't covered by tests 😕

[the remark this answers is not in the document]
Comment (aboedo) [reply Purchases/Error Handling/StoreKitError+Extensions.swift:38]: could you create a jira task for it? it feels like a bad thing to potentially break without noticing

[the remark this answers is not in the document]
Comment (NachoSoto) [reply Purchases/Error Handling/StoreKitError+Extensions.swift:38]: [CF-323]

[the remark this answers is not in the document]
Comment (aboedo) [reply Purchases/Error Handling/StoreKitError+Extensions.swift:38]: awesome, thanks!

**Pull request merged**